github/resource_github_team_sync_group_mapping.go

```diff
@@ -0,0 +1,210 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGithubTeamSyncGroupMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubTeamSyncGroupMappingCreate,
+		Read:   resourceGithubTeamSyncGroupMappingRead,
+		Update: resourceGithubTeamSyncGroupMappingUpdate,
+		Delete: resourceGithubTeamSyncGroupMappingDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("team_slug", d.Id())
+				d.SetId(fmt.Sprintf("teams/%s/team-sync/group-mappings", d.Id()))
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"team_slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
```

Comment:
@anGie44 I know this has been long part of the provider, but what was the reasoning for this? At first sight, it doesn't seem like an update to the groups should require a new resource. Why wasn't in-place updating possible?